engine/cdn/redis/line.go

```diff
@@ -8,9 +8,10 @@ import (
 )
 
 type Line struct {
-	Number int64  `json:"number"`
-	Value  string `json:"value"`
-	Since  int64  `json:"since,omitempty"`
+	Number     int64  `json:"number"`
+	Value      string `json:"value"`
+	ApiRefHash string `json:"api_ref_hash"`
```

Comment:
This field is only used when listening /stream endpoint. Maybe the websocket should send data with a dedicated line struct and set api_ref_hash value just before sending a line.
Also this will prevent from giving the full Item to the Redis reader. As this data is not stored in Redis is should not be set on lines inside Redis package.